pkg/restapi/operation/didexchange/didexchange.go

```diff
@@ -350,21 +351,66 @@ func (c *Operation) startClientEventListener() error {
 		return fmt.Errorf("didexchange message event registration failed: %w", err)
 	}
 
-	// auto execute the actions
+	// event listeners
 	go func() {
-		err := service.AutoExecuteActionEvent(c.actionCh)
-		if err != nil {
-			logger.Errorf("auto action event execution failed: %s", err)
+		for {
+			select {
+			case e := <-c.actionCh:
+				c.handleActionEvents(e)
+			case e := <-c.msgCh:
+				c.handleMessageEvents(e)
+			}
 		}
 	}()
 
-	go func() {
-		for e := range c.msgCh {
-			// TODO https://github.com/hyperledger/aries-framework-go/issues/551 - Integrate Message event with Webhook
-			// for now, log the messages
-			logger.Infof("message event received : type=%s", e.Type)
+	return nil
+}
+
+func (c *Operation) handleMessageEvents(e service.StateMsg) {
+	if e.Type == service.PostState {
+		var props didexchange.Event
+		switch v := e.Properties.(type) {
+		case didexchange.Event:
+			props = v
+		case error:
+			logger.Warnf("service processing failed : %s", v)
+			return
+		default:
+			logger.Errorf("event is not of DIDExchange event type")
+			return
 		}
-	}()
+
+		err := c.sendConnectionNotification(props.ConnectionID(), e.StateID)
+		if err != nil {
+			logger.Errorf("handleMessageEvents - send connection notification failed : %s", err)
+		}
+	}
+}
+
+func (c *Operation) handleActionEvents(e service.DIDCommAction) {
+	// TODO https://github.com/hyperledger/aries-framework-go/issues/579 - Integrate Action events with webhooks
+	e.Continue()
+}
+
+func (c *Operation) sendConnectionNotification(connectionID, stateID string) error {
+	// TODO - fetch connection data from DB (update once https://github.com/hyperledger/aries-framework-go/pull/573
+	//  is merged)
+	conn := &didexchange.Connection{
+		ConnectionRecord: didexsvc.ConnectionRecord{
+			ConnectionID: connectionID,
+			State:        stateID,
+		},
+	}
+
+	jsonMessage, err := json.Marshal(conn)
+	if err != nil {
+		return fmt.Errorf("connection notification json marshal : %w", err)
+	}
+
+	err = c.notifier.Notify("connections", jsonMessage)
```

Comment:
I think that `"connections"` should be a constant somewhere since it'll be used in quite a few places. Perhaps a constant in the payload.go file?

Reply:
https://github.com/hyperledger/aries-framework-go/pull/580#discussion_r338765349